content/ko/docs/tasks/administer-cluster/configure-upgrade-etcd.md

```diff
@@ -0,0 +1,409 @@
+접근 권한을 갖는 것이 바람직하다. 데이터의 민감도를 고려하여,
+etcd 클러스터에 대한 접근이 필요한 노드에만 권한을 부여하는
+것이 권장된다.
+
+etcd를 보호하려면 방화벽 규칙을 설정하거나 etcd에서 제공하는 보안
+기능을 사용한다. etcd 보안 기능은 x509 공개키 인프라스트럭처(PKI)에
+따라서 결정된다. 시작하려면 키와 인증서 쌍을 생성하여 보안 통신
+채널을 설정한다. 예를 들어, etcd 구성원 간의 보안 통신을 위해
+`peer.key` 및 `peer.cert` 키 쌍을 사용하고, etcd와 해당 클라이언트
+간의 보안 통신을 위해 `client.key` 와 `client.cert` 키 쌍을 사용한다.
+클라이언트 인증을 위한 키 쌍과 CA 파일을 생성하려면 etcd 프로젝트에서
+제공하는 [예제 스크립트](https://github.com/coreos/etcd/tree/master/hack/tls-setup)를
+참고한다.
+
+### 보안 통신
+
+보안 피어 통신으로 etcd를 구성하려면,
+`--peer-key-file=peer.key` 및 `--peer-cert-file=peer.cert` 플래그를 지정하고 URL
+스키마로 HTTPS를 사용한다.
+
+마찬가지로, 보안 클라이언트 통신으로 etcd를 구성하려면,
+`--key-file=k8sclient.key` 및 `--cert-file=k8sclient.cert` 플래그를 지정하고 URL
+스키마로 HTTPS를 사용한다. 다음은 보안 통신을 사용하는 클라이언트 명령어의
+예시이다.
+
+```
+ETCDCTL_API=3 etcdctl --endpoints 10.2.0.9:2379 \
+  --cert=/etc/kubernetes/pki/etcd/server.crt \
+  --key=/etc/kubernetes/pki/etcd/server.key \
+  --cacert=/etc/kubernetes/pki/etcd/ca.crt \
+  member list
+```
+
+### etcd 클러스터의 접근 제한
+
+보안 통신 구성을 마치면, etcd 클러스터의 접근을 쿠버네티스
+API 서버로만 제한한다. 이를 위해 TLS 인증을 사용한다.
+
+CA `etcd.ca`가 신뢰하는 키 쌍 `k8sclient.key` 와 `k8sclient.cert`를
+예로 들어본다. etcd가 TLS와 함께 `--client-cert-auth`로 구성되면,
+시스템 CA나 `--trusted-ca-file` 플래그로 전달된 CA를 사용하여
+클라이언트의 인증서를 확인한다.
+`--client-cert-auth=true` 와 `--trusted-ca-file=etcd.ca` 플래그를 지정하면
+`k8sclient.cert` 인증서를 가진 클라이언트에 대한 접근이 제한된다.
+
+etcd가 올바르게 구성되면 유효한 인증서를 가진 클라이언트만 엑세스할 수
+있다. 쿠버네티스 API 서버에 접근 권한을 부여하려면 `--etcd-certfile=k8sclient.cert`,
+`--etcd-keyfile=k8sclient.key` 및 `--etcd-cafile=ca.cert` 플래그를
+사용하여 서버를 구성한다.
+
+{{< note >}}
+etcd 인증은 현재 쿠버네티스에서 지원되지 않는다. 자세한
+내용은 관련된 이슈
+[Etcd v2에 대한 기본 인증 지원](https://github.com/kubernetes/kubernetes/issues/23398)을 참고한다.
+{{< /note >}}
+
+## 실패한 etcd 멤버 교체
+
+etcd 클러스터는 사소한 멤버 장애를 허용함으로써 고가용성을 달성한다.
+그러나 클러스터 전반의 상태를 개선하려면 장애가 발생한 멤버를 즉시
+교체해야 한다. 여러 구성원이 실패하는 경우엔, 하나씩 교체한다.
+장애 멤버 교체는 장애 멤버를 제거하고 새 멤버를 추가하는 두 단계가
+포함된다.
+
+etcd는 내부적으로 고유한 멤버 ID를 사용하고 있지만, 사람에 의한 실수를
+방지하기 위해 각 멤버에 고유한 이름을 사용하는 것을 권장한다. 예를 들어,
+세 명의 멤버로 구성된 etcd 클러스터 가정해 보자. `member1=http://10.0.0.1`,
+`member2=http://10.0.0.2`, 그리고 `member3=http://10.0.0.3`의 URL을 사용한다.
+`member1`이 실패하면 `member4=http://10.0.0.4`로 대체한다.
+
+1. 실패한 `member1`의 멤버 ID를 가져온다.
+
+   ```shell
+   etcdctl --endpoints=http://10.0.0.2,http://10.0.0.3 member list
+   ```
+
+   다음과 같은 메시지가 표시된다.
+
+   ```console
+   8211f1d0f64f3269, started, member1, http://10.0.0.1:2380, http://10.0.0.1:2379
+   91bc3c398fb3c146, started, member2, http://10.0.0.2:2380, http://10.0.0.2:2379
+   fd422379fda50e48, started, member3, http://10.0.0.3:2380, http://10.0.0.3:2379
+   ```
+
+1. 다음 중 하나를 수행한다.
+
+   1. 각 쿠버네티스 API 서버가 모든 etcd 멤버와 통신하도록 구성된 경우,
+      `--etcd-servers` 플래그에서 실패한 멤버를 제거한 다음
+      각 쿠버네티스 API 서버를 재시작한다.
+   1. 각 쿠버네티스 API 서버가 단일 etcd 멤버와 통신하는 경우,
+      실패한 etcd와 통신하는 쿠버네티스 API 서버를
+      중지한다.
+
+1. 장애가 발생한 노드의 etcd 서버를 중지한다. 쿠버네티스 API 서버
+   이외의 다른 클라이언트가 etcd로 트래픽을 발생시킬 수 있으므로,
+   데이터 디렉터리에 쓰기를 방지하기 위해 모든 트래픽을 중지하는 것이
```

Comment:
[원문]
> Stop the etcd server on the broken node. It is possible that other clients besides the Kubernetes API server is causing traffic to etcd and it is desirable to stop all traffic to prevent writes to the data dir.

좀 더 자연스럽도록 부연 설명을 해 보았습니다.

```suggestion
1. 장애가 발생한 노드의 etcd 서버를 중지한다. 쿠버네티스 API 서버
   이외의 다른 클라이언트가 etcd로 트래픽을 발생시킬 수 있으므로,
   데이터 디렉터리에 데이터 쓰기 작업이 일어나는 것을 방지하기 위해 모든 트래픽을 중지하는 것이
```

Reply:
라인 정리 포함하여 적용하겠습니다. 리뷰 감사합니다!